### ISSUE TITLE
Async function return value inlay hint error

### ISSUE THREAD

**Issue by yanyongyu** [microsoft/pyright#5639]
## Environment data

-   Language Server version: 2023.8.10
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

## Code Snippet

```python
async def test():
    pass
```

## Repro Steps

Pylance report the return value as `Coroutine[Any, Any, None]` in inlay hint:

```python
async def test() -> Coroutine[Any, Any, None]:
    pass
```

## Expected behavior

```python
async def test() -> None:
    pass
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.321, which I just published. It will also be included in a future release of pylance.
